### PR TITLE
Update atom densities feature calculation

### PR DESCRIPTION
**Pull request by CunliangGeng**
To solve issue #92.
The new code calculates atom densities based on elements. The default is C, N, O and S. Thus both  backbone and sidechain are used for feature calculation.

NOTE: **Please merge this RP #101  after merging the RP #98**

### CODE REVIEW

File: deeprank/config/chemicals.py [hunk 1]
```diff
@@ -70,4 +70,17 @@
     'THR': 'polar',
     'TRP': 'polar',
     'TYR': 'polar'
-    }
\ No newline at end of file
+    }
+
+
+# atom vdw radius
+# https://en.wikipedia.org/wiki/Van_der_Waals_radius
```
Comment (NicoRenaud) [inline deeprank/config/chemicals.py]: Could we use something else than Wikipedia ?
The python package Mendeleev has vdw radius for all atoms I think

File: deeprank/generate/DataGenerator.py [hunk 2]
```diff
@@ -802,7 +803,7 @@ def map_features(self, grid_info={},
         >>> grid_info = {
         >>>     'number_of_points' : [30,30,30],
         >>>     'resolution' : [1.,1.,1.],
-        >>>     'atomic_densities' : {'CA':3.5,'N':3.5,'O':3.5,'C':3.5},
+        >>>     'atomic_densities' : {'C':1.7, 'N':1.55, 'O':1.52, 'S':1.8},
```
Comment (NicoRenaud) [inline deeprank/generate/DataGenerator.py]: Why half of the radius ? Did I use the diameter ? There also we could use Mendeleev ...

File: deeprank/generate/GridTools.py [hunk 3]
```diff
@@ -35,8 +35,8 @@ def __init__(self, molgrp,
             number_of_points(int, optional): number of points we want in
                 each direction of the grid.
             resolution(float, optional): distance(in Angs) between two points.
-            atomic_densities(dict, optional): dictionary of atom types with
-                their vdw radius, e.g. {'CA':1.7, 'C':1.7, 'N':1.55, 'O':1.52}
+            atomic_densities(dict, optional): dictionary of element types with
```
Comment (NicoRenaud) [inline deeprank/generate/GridTools.py]: atomType is the default name in the PDB file format no ? I have nothing against elementtype but I don't get why changing it

File: deeprank/tools/pdb2sql.py [hunk 4]
```diff
@@ -113,7 +113,8 @@ def _create_sql(self):
                     'y': 'REAL',
                     'z': 'REAL',
                     'occ': 'REAL',
-                    'temp': 'REAL'
+                    'temp': 'REAL',
+                    'element': 'TEXT'
```
Comment (NicoRenaud) [inline deeprank/tools/pdb2sql.py]: I don't know what this element is but we should also use it in pdb2sql then ... and also use pdb2sql in deeprank :)

File: deeprank/config/chemicals.py (hunk 1 above)
Comment (CunliangGeng) [reply deeprank/config/chemicals.py]: We need vdw_radius for only a few elements, maybe too heavy to import a package to do that.
Changed the reference to a book.

File: deeprank/generate/DataGenerator.py (hunk 2 above)
Comment (CunliangGeng) [reply deeprank/generate/DataGenerator.py]: The Mendeleev has the same VDW values as what I'm using.

File: deeprank/generate/GridTools.py (hunk 3 above)
Comment (CunliangGeng) [reply deeprank/generate/GridTools.py]: The new code are calculating atomic densities based on element types (C,N,O,S) but not the previous atom types (e.g. CA, CB, C, N, O). Thus the new code will cover both backbone and sidechain atoms.  
Using element type is a convenient way to consider side chain atoms (which has ~ 40 atom types).

File: deeprank/tools/pdb2sql.py (hunk 4 above)
Comment (CunliangGeng) [reply deeprank/tools/pdb2sql.py]: The element here is chemical element.

File: deeprank/generate/GridTools.py (hunk 3 above)
Comment (NicoRenaud) [reply deeprank/generate/GridTools.py]: sounds good

File: deeprank/tools/pdb2sql.py (hunk 4 above)
Comment (NicoRenaud) [reply deeprank/tools/pdb2sql.py]: using pdb2sql will maybe not work out of the box as some functionalities ma be slightly different ... so be careful there

**Comment by NicoRenaud** [comment]
If you fix the conflicts it's all good for me and we can merge

**Pull request merged**